v2/processor.go

```diff
@@ -81,24 +94,73 @@ func NewProcessor(receiver Receiver, handler HandlerFunc, options *ProcessorOpti
 			opts.StartRetryDelayStrategy = options.StartRetryDelayStrategy
 		}
 	}
+	return opts
+}
+
+func NewProcessor(receiver Receiver, handler HandlerFunc, options *ProcessorOptions) *Processor {
+	opts := applyProcessorOptions(options)
+	receiverEx := NewReceiverEx("receiver", receiver)
+	return &Processor{
+		receivers:         map[string]*ReceiverEx{receiverEx.name: receiverEx},
+		handle:            handler,
+		options:           *opts,
+		concurrencyTokens: make(chan struct{}, opts.MaxConcurrency),
+	}
+}
+
+func NewMultiProcessor(receiversEx []*ReceiverEx, handler HandlerFunc, options *ProcessorOptions) *Processor {
+	opts := applyProcessorOptions(options)
+	var receivers = make(map[string]*ReceiverEx)
+	for _, receiver := range receiversEx {
+		receivers[receiver.name] = receiver
+	}
 	return &Processor{
-		receiver:          receiver,
+		receivers:         receivers,
 		handle:            handler,
-		options:           opts,
+		options:           *opts,
 		concurrencyTokens: make(chan struct{}, opts.MaxConcurrency),
 	}
 }
 
-// Start starts the processor and blocks until an error occurs or the context is canceled.
+// Start starts processing on all the receivers of the processor and blocks until all processors are stopped or the context is canceled.
+// It will retry starting the processor based on the StartMaxAttempt and StartRetryDelayStrategy.
+// Returns a combined list of errors encountered during each processor start.
+func (p *Processor) Start(ctx context.Context) error {
+	wg := sync.WaitGroup{}
+	errChan := make(chan error, len(p.receivers))
+	for name := range p.receivers {
+		wg.Add(1)
```

Comment:
Here, if startOne panics, we're not entering wg.Done() because we don't recover from it.
so I think that the wg.Wait() will keep us stuck forever.

could you try to add a test for that by generating a panic somehow inside startOne ?

Reply:
in terms of fixing it, I've been itching to use conc library in go-shuttle, but unsure whether that's wise as a library.
we should handle the panic I think...

Reply:
added panic handling + UTs